remoting/zookeeper/client.go

```diff
@@ -285,6 +285,11 @@ func (z *ZookeeperClient) HandleZkEvent(session <-chan zk.Event) {
 				if state == (int)(zk.StateHasSession) {
 					continue
 				}
+				if event.State == zk.StateHasSession {
+					atomic.StoreUint32(&z.valid, 1)
+					close(z.reconnectCh)
+					z.reconnectCh = make(chan struct{})
```

Comment:
I think u need to use a lock to promise that: 

```go
    close(z.reconnectCh)
    z.reconnectCh = make(chan struct{})
```

Reply:
There only have one goroutine run this function, don't need lock..